cibuildwheel/__main__.py

```diff
@@ -281,6 +281,8 @@ def main() -> None:
     if not output_dir.exists():
         output_dir.mkdir(parents=True)
 
+    existing_contents = set(output_dir.iterdir())
```

Comment:
Good catch; I hadn't thought of there already being wheels present. What about `output_dir.glob("*.whl")`, since you're saying "produced wheels"?

Reply:
Well, we shouldn't be producing non-wheels, and if we do, we don't want to hide it, that would make debugging harder?

Reply:
There's something to be said about that. So if that's the case, we might even want to check that all produced files end in `.whl`?

Reply:
But yeah, nothing urgent either.

Reply:
It shouldn't be common (or possible?), but I'd currently just like not to hide it if it happens, rather than erroring out quite yet on it happening.

The thing that would help me is if you could detect if the git dir was dirty. That causes badly named wheels when using setuptools_scm, which can't upload to PyPI.